a10_octavia/controller/worker/tasks/cert_tasks.py

```diff
@@ -63,143 +64,153 @@ class SSLKeyCreate(task.Task):
-            pass
+        if cert_data is not None:
+            try:
+                self.axapi_client.file.ssl_key.delete(file=cert_data.key_filename)
+                LOG.debug("Reverted SSLKeyCreate: %s", cert_data.key_filename)
+            except Exception:
+                pass
 
 
 class ClientSSLTemplateCreate(task.Task):
     """Task to create a client ssl template for a listener"""
     @axapi_client_decorator
     def execute(self, cert_data, vthunder):
-        try:
-            if self.axapi_client.slb.template.client_ssl.exists(name=cert_data.template_name):
-                self.axapi_client.slb.template.client_ssl.update(name=cert_data.template_name,
-                                                                 cert=cert_data.cert_filename,
-                                                                 key=cert_data.key_filename,
-                                                                 passphrase=cert_data.key_pass)
-            else:
-                self.axapi_client.slb.template.client_ssl.create(name=cert_data.template_name,
-                                                                 cert=cert_data.cert_filename,
-                                                                 key=cert_data.key_filename,
-                                                                 passphrase=cert_data.key_pass)
-        except Exception as e:
-            LOG.exception("Failed to create client SSL template: %s", str(e))
-            raise e
+        if cert_data is not None:
+            try:
+                if self.axapi_client.slb.template.client_ssl.exists(name=cert_data.template_name):
+                    self.axapi_client.slb.template.client_ssl.update(name=cert_data.template_name,
+                                                                     cert=cert_data.cert_filename,
+                                                                     key=cert_data.key_filename,
+                                                                     passphrase=cert_data.key_pass)
+                else:
+                    self.axapi_client.slb.template.client_ssl.create(name=cert_data.template_name,
+                                                                     cert=cert_data.cert_filename,
+                                                                     key=cert_data.key_filename,
+                                                                     passphrase=cert_data.key_pass)
+            except Exception as e:
+                LOG.exception("Failed to create client SSL template: %s", str(e))
+                raise e
 
     @axapi_client_decorator
     def revert(self, cert_data, vthunder, *args, **kwargs):
-        try:
-            self.axapi_client.slb.template.client_ssl.delete(name=cert_data.template_name)
-            LOG.debug("Reverted ClientSSLTemplateCreate: %s", cert_data.template_name)
-        except Exception:
-            pass
+        if cert_data is not None:
```

Comment:
This shouldn't be needed. We've designed the flows so that cert_data will never be none if this task is executed.